### PR TITLE
crates.io release prep

### PR DESCRIPTION
**Pull request by dignifiedquire**
- [x] update version deps
- [x] configure `cargo-release`
- [x] add badges to `README.md`s

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+```sh
+$ cargo release <patch|minor|major>
+```
+
+Actual publishing
+
+```sh
+$ cargo release --execute <patch|minor|major>
+```
```
Comment (b5) [inline RELEASE.md:39]: I'm assuming this requires special credentials? Should be noted if so.

File: examples/Cargo.toml [hunk 2]
```diff
@@ -15,10 +16,10 @@ bytesize = "1.1.0"
...
 futures = "0.3.21"
 indicatif = "0.17.0"
-iroh-car = { path = "../iroh-car" }
-iroh-resolver = { path = "../iroh-resolver" }
-iroh-rpc-client = { path = "../iroh-rpc-client" }
-iroh-util = { path = "../iroh-util" }
+iroh-car = { version = "0.1.1", path = "../iroh-car" }
+iroh-resolver = { version = "0.1.1", path = "../iroh-resolver" }
+iroh-rpc-client = { version = "0.1.1", path = "../iroh-rpc-client" }
+iroh-util = { version = "0.1.1", path = "../iroh-util" }
```
Comment (b5) [inline examples/Cargo.toml:22]: aren't we doing "0.1.2", or is this an artifact of being the first release?

File: iroh-api/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# iroh-api
+
+[![crates.io](https://img.shields.io/crates/v/iroh-api.svg?style=flat-square)](https://crates.io/crates/iroh-api)
+[![Released API docs](https://img.shields.io/docsrs/iroh-api?style=flat-square)](https://docs.rs/iroh-api)
+[![MIT/Apache-2.0 licensed](https://img.shields.io/crates/l/iroh-api?style=flat-square)](../LICENSE-MIT)
+[![CI](https://img.shields.io/github/workflow/status/n0-computer/iroh/Continuous%20integration?style=flat-square)](https://github.com/n0-computer/iroh/actions?query=workflow%3A%22Continuous+integration%22)
```
Comment (b5) [inline iroh-api/README.md:6]: can confirm these links actually work 😄

File: examples/Cargo.toml (hunk 2 above)
Comment (dignifiedquire) [reply examples/Cargo.toml:22]: this is just noting the current version, not the new one

File: RELEASE.md (hunk 1 above)
Comment (dignifiedquire) [reply RELEASE.md]: yes, added

**Review by b5 — APPROVED** [review]
I've smoke tested this branch, looks good!

**Pull request merged**